src/rebalancer.rs

```diff
@@ -365,41 +361,43 @@ impl Rebalancer {
             RequirementType::Equity,
             BalanceSide::Liabilities,
         )?;
-        if liab_usd_value < self.config.token_account_dust_threshold {
+        if liab_usd_value < self.token_account_dust_threshold {
             thread_debug!(
                 "The {:?} unscaled Liability tokens of Bank {:?} are below the dust threshold {}.",
                 liab_balance,
                 bank_pk,
-                self.config.token_account_dust_threshold
+                self.token_account_dust_threshold
             );
             return Ok(());
         }
 
         thread_debug!(
-            "The {:?} unscaled Liability tokens of Bank {:?} need to be repaid with {:?} USD.",
+            "The {:?} unscaled Liability tokens of the Bank {:?} need to be repaid with {:?} tokens of the preferred asset Bank {:?}, Mint {:?}.",
             liab_balance,
             bank_pk,
-            liab_usd_value
+            liab_usd_value,
+            self.swap_mint_bank,
+            self.swap_mint
         );
 
         // Get the amount of swap token needed to repay the liability
-        let required_swap_token = self.get_amount(
-            liab_usd_value,
-            &self.swap_mint_bank_pk,
-            Some(PriceBias::Low),
-        )?;
+        let required_swap_token =
+            self.get_amount(liab_usd_value, &self.swap_mint_bank, Some(PriceBias::Low))?;
 
         let withdraw_amount = if required_swap_token.is_positive() {
-            let (max_withdraw_amount, withdraw_all) =
-                self.get_max_withdraw_for_bank(&self.swap_mint_bank_pk, lq_account)?;
+            // The withdral_all flag is purposefully ignored here.
```

Comment:
The fix to prevent Eva from occasionally draining the preferred asset.

Reply:
I think we should still dig into why it is even trying to withdraw a preferred deposit. But in any case this is for another PR. This one has been growing for too long already, let's merge it in!

Reply:
That was an attempt to use the preferred token to pay the newly acquired liability, which is the way it supposed to be, I believe.

Reply:
Ah, true, then we probably should move this step after the other ones, so that we always have enough collateral. But again, for the next PR, this looks good!

Reply:
Just one little typo:
```suggestion
            // The withdraw_all flag is purposefully ignored here.
```

Reply:
> Ah, true, then we probably should move this step after the other ones, so that we always have enough collateral. But again, for the next PR, this looks good!

I envision that we will eventually revising the way the liquidation/rebalancing transactions are being created. Some of them could be combined.